components/workspace/WorkspaceAppearancePicker.tsx

```
@@ -0,0 +1,218 @@
+        const parsed = JSON.parse(saved);
+        return parsed.filter((c: string) => !WORKSPACE_THEME_COLORS.includes(c));
+      }
+    } catch (error) {
+      console.error("Failed to load custom colors", error);
+    }
+    return [];
+  });
+
+  useEffect(() => {
+    if (themeColorValue) {
+      handleCustomColorAdd(themeColorValue);
```

ESLint

[error] Error: Cannot access variable before it is declared

`handleCustomColorAdd` is accessed before it is declared, which prevents the earlier access from updating when this value changes over time.

/home/jailuser/git/components/workspace/WorkspaceAppearancePicker.tsx:45:7
  43 |   useEffect(() => {
  44 |     if (themeColorValue) {
> 45 |       handleCustomColorAdd(themeColorValue);
     |       ^^^^^^^^^^^^^^^^^^^^ `handleCustomColorAdd` accessed before it is declared
  46 |     }
  47 |     // eslint-disable-next-line react-hooks/exhaustive-deps
  48 |   }, []);

/home/jailuser/git/components/workspace/WorkspaceAppearancePicker.tsx:50:3
  48 |   }, []);
  49 |
> 50 |   const handleCustomColorAdd = (color: string) => {
     |   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> 51 |     onThemeColorChange(color);
     | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> 52 |     if (!WORKSPACE_THEME_COLORS.includes(color) && !customColors.includes(color)) {
     …
     | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> 60 |     }
     | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> 61 |   };
     | ^^^^^ `handleCustomColorAdd` is declared here
  62 |
  63 |   const handleCustomColorRemove = (colorToRemove: string) => {
  64 |     const updated = customColors.filter((c) => c !== colorToRemove);

(react-hooks/immutability)